logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2/threadinfo/ApplicationThread.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.logging.log4j2.threadinfo;
+
+import org.apache.logging.log4j.ThreadContext;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+public class ApplicationThread extends Thread {
+    private final User user;
+
+    public ApplicationThread(User user) {
+        this.user = user;
+        user.connect();
+    }
+
+    @Override
+    public void run() {
+        ThreadContext.put("name", user.getName());
+        BooleanSupplier poll;
+
+        do {
+            poll = user.actionPoll();
+
+            sleepRandom();
+        } while (poll == null || poll.getAsBoolean());
```

Comment:
I can't help but feel that this is a bit complex for an example. I see what you're doing, but you're mixing the actions, which are really `Runnable` with the terminate condition, which is really a `keepGoing -> true/false`

I like the action list, but I think the code would be easier to understand if you made the action list of `Runnable` and set a flag for whether the user should stop running.